src/fragments/lib/datastore/js/data-access/update-snippet.mdx

```diff
@@ -1,11 +1,12 @@
 ```js
-const original = await DataStore.query(Post, "123");
-
-await DataStore.save(
-  Post.copyOf(original, updated => {
-    updated.title = `title ${Date.now()}`;
-  })
-);
+async function updatePost(id, newTitle) {
+  const original = await DataStore.query(Post, id);
+  await DataStore.save(
+    Post.copyOf(original, updated => {
+      updated.title = newTitle
+    })
+  );
+}
```

Comment:
If the app is not subscribing to real time events, and had performed a query for a list of models, and end-user update one of the models, app calls DataStore.save(), then it should get the latest local model and replace fields that is being updated, title in this case

Reply:
would this example conflict with the following example of performing `observeQuery` and maintaining the latest local data in memory? Since with the real time APIs, they don't need to query before saving

Reply:
Yeah, I'm not so sure. Not every use-case will call for live data. The aim is just to cover both use cases.